files/default/neutron-ha-tool.py

```diff
@@ -559,29 +581,108 @@ def migrate_router(qclient, router_id, agent_id, target_id):
     """
 
     LOG.info("Migrating router=%s from agent=%s to agent=%s",
-             router_id, agent_id, target_id)
+             router_id, agent['id'], target['id'])
 
     # N.B. The neutron API will return "success" even when there is a
     # subsequent failure during the add or remove process so we must check to
     # ensure the router has been added or removed
 
     # Remove the router from the original agent
-    qclient.remove_router_from_l3_agent(agent_id, router_id)
-    LOG.debug("Removed router from agent=%s" % agent_id)
+    qclient.remove_router_from_l3_agent(agent['id'], router_id)
+    LOG.debug("Removed router from agent=%s" % agent['id'])
 
     # ensure it is removed or log an error
-    if router_id in list_routers_on_l3_agent(qclient, agent_id):
+    if router_id in list_routers_on_l3_agent(qclient, agent['id']):
         raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
 
     # add the router id to a live agent
     router_body = {'router_id': router_id}
-    qclient.add_router_to_l3_agent(target_id, router_body)
+    qclient.add_router_to_l3_agent(target['id'], router_body)
 
     # ensure it is removed or log an error
-    if router_id not in list_routers_on_l3_agent(qclient, target_id):
+    if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
-                           (router_id, agent_id))
+                           (router_id, agent['id']))
+    if wait_for_ports:
+        wait_router_ports_active(qclient, router_id, target['host'])
+
+    if delete_namespace:
+        ssh_delete_router_namespace(agent['host'], router_id)
```

Comment:
Sure. See the ```ssh_delete_router_namespace``` function:  https://github.com/SUSE-Cloud/cookbook-openstack-network/pull/8/files#diff-937ed0b9cba9efd2521c8fd1c9e8416fR666 ;)